Test update AK with syspurpose values

swadeley opened the pull request:
Hello

this branch depends on (must not be merged before):
Add test create AK with syspurpose values (#7578)

Thank you

mirekdlugosz commented on tests/foreman/cli/test_activationkey.py:
```diff
@@ -1149,6 +1150,122 @@ def test_positive_remove_host_collection_by_name(self):
+        self.assertEqual(new_ak['system-purpose']['purpose-addons'], '')
+        self.assertEqual(new_ak['system-purpose']['purpose-role'], '')
+        self.assertEqual(new_ak['system-purpose']['purpose-usage'], '')
+        self.assertEqual(new_ak['system-purpose']['service-level'], '')
```
Do we need that? Is there possibility that uploading manifest will change these values for existing activation key?

swadeley replied:
Adding the manifest makes those values available in the system level drop-down menu.
That does not mean they could be set due to some bug.  I can remove that.

swadeley commented:
> I kind of see where you are going, but I can't help but notice that there is large overlap between this new test and `test_create_ak_with_syspurpose_set`. Is defending against manifest upload error / ensuring manifest upload doesn't interact with that workflow really worth the effort? Especially as running Satellite without manifest is not really that common for customers (I assume).
@mirekdlugosz 
The issue is that the values such as Self-Support and Premium do not appear until you have a manifest. I only found this out yesterday during testing for this PRs test. But I do not really need my own org, and do not need repos or a product.
> 
> Second thing - this PR doesn't build on top of #7578, it includes #7578. You need to either:
> 
>     1. Drop [527dc83](https://github.com/SatelliteQE/robottelo/commit/527dc83398019a2258010ee931b0cea9fc79bc6f) from this PR. Basically `git rebase -i origin/master` and change first `pick` to `drop`.
> 
>     2. Close #7578 as being superseded by this one.

Yes, because I created this tests & PRs new branch from the branch I used for #7578  as I needed the  changes in robottelo/cli/factory.py  to make it work.

I had hoped someone would approve and merge #7578 before today's post snap test run (then i could have re-based from master). As that is probably not going to happen I could just do as you suggest and close #7578 as being superseded by this one.

thank you

mirekdlugosz commented:
> The issue is that the values such as Self-Support and Premium do not appear until you have a manifest. I only found this out yesterday during testing for this PRs test. But I do not really need my own org, and do not need repos or a product.

What I was trying to say, is that basically we have:
* `test_create_ak_with_syspurpose_set`
    1. create new activation key with sys purpose fields
    2. update this AK to remove content of sys purpose
* `test_update_ak_with_syspurpose_values`
    1. create new activation key without sys purpose fields
    2. upload manifest
    3. update this AK to have some sys purpose fields
    4. update this AK to some other sys purpose fields

In my mind, these two are similar enough that you don't need two tests. You could probably go away with:
1. upload manifest
2. create new AK without sys purpose fields
3. update this AK to have sys purpose fields
4. update it again to not have sys purpose fields

swadeley commented:
@mirekdlugosz and @omkarkhatavkar 

I have changed the test to use the class org and added the fake manifest step to the class setup. That way other tests can use cls org if they need fake manifest (I will make a note to review the other tests in tests/foreman/cli/test_activationkey.py for this).

I decided to make separate git commits for ease of working on the two branches (for this pull request and #7578)

Thank you

swadeley commented:
<snip>
> 
> In my mind, these two are similar enough that you don't need two tests. You could probably go away with:
> 
>     1. upload manifest
> 
>     2. create new AK without sys purpose fields
@mirekdlugosz 
If I only had that test then I would not have found this bug:
https://bugzilla.redhat.com/show_bug.cgi?id=1789028
> 
>     3. update this AK to have sys purpose fields
> 
>     4. update it again to not have sys purpose fields
> 
> 
> Or some other variation of that.

My tests are based on the user stories I found here:
https://projects.theforeman.org/issues/24180

I have now moved the manifest upload to the cls setup to save time for the two tests I am working on (and, in time, other tests can be updated to use this).

I of course respect your opinion. Let me know if this seems acceptable reasoning.

Thank you

swadeley commented:
robottelo]$ pytest tests/foreman/cli/test_activationkey.py -k test_update_ak_with_syspurpose_values
==================== test session starts ========================================
platform linux -- Python 3.7.5, pytest-4.6.3, py-1.8.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: cov-2.8.1, services-1.3.1, xdist-1.30.0, mock-1.10.4, forked-1.1.3
collecting ... 2020-01-14 10:00:20 - conftest - DEBUG - Collected 54 test cases
2020-01-14 10:00:21 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f1d00f30350
2020-01-14 10:00:21 - robottelo.ssh - INFO - Connected to [dell-r330-12.gsslab.brq.redhat.com]
2020-01-14 10:00:21 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-01-14 10:00:22 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-4.beta.el7sat.noarch

2020-01-14 10:00:22 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f1d00f30350
2020-01-14 10:00:22 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-01-14 10:00:22 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 54 items / 53 deselected / 1 selected                                                                                                                             

tests/foreman/cli/test_activationkey.py .                                                                                       [100%]
================== 1 passed, 53 deselected in 53.81 seconds ==========================

swadeley commented:
> @swadeley If you want to have it in two tests, that's fine for me. I definitely won't block PR based on reasonable disagreement.
> 
> It still needs Tier2 reviewer ACK to merge.
> 
> Also, I see both this and #7578 has the same changes. I would suggest to remove "DO NOT MERGE" label from one of them, wait for it to be merged and rebase other one on top of master.

@mirekdlugosz thank you, I have requested a review of  #7578 from @omkarkhatavkar . Requesting that one gets merged before this one. Then will rebase.

omkarkhatavkar commented:
@swadeley done ! you should be good

swadeley commented:
Thank you all
